modules/reindex/src/yamlRestTest/resources/rest-api-spec/test/reindex/30_search.yml

```diff
@@ -152,6 +152,10 @@
   - set: {task: task}
 
   - do:
+      # Inherits the warning from the previous task
+      warnings:
+        - The sort option in reindex is deprecated. Instead consider using query
```

Comment:
It looks like with the latest change `getTask` now inherits the warnings from the previous task which is technically is a breaking change :(

Reply:
I suspect this is because we call the `RemovedTaskListener` (and therefore complete the future) in the thread context of the task that's being removed. We need to fix that, we can't just leak context across responses like this.